File: FileHandler/Python/FileHandler.py
class FileHandler:
    def __init__(self, file_path):
        self.file_path = file_path

    def open_file(self, mode='r'):
        try:
            self.file = open(self.file_path, mode)
            print(f"File {self.file_path} opened in {mode} mode.")
        except Exception as e:
            print(f"Error opening file: {e}")
            self.file = None

    def read_file(self):
        if self.file and not self.file.closed:
            try:
                content = self.file.read()
                return content
            except Exception as e:
                print(f"Error reading file: {e}")
                return None
        else:
            print("File is not open.")
            return None

    def write_file(self, content):
        if self.file and not self.file.closed and self.file.mode in ['w', 'a', 'r+']:
            try:
                self.file.write(content)
                print("Content written to file.")
            except Exception as e:
                print(f"Error writing to file: {e}")
        else:
            print("File is not open in write mode.")

    def close_file(self):
        if self.file and not self.file.closed:
            try:
                self.file.close()
                print(f"File {self.file_path} closed.")
            except Exception as e:
                print(f"Error closing file: {e}")

File: FileHandler/FileHandler.cpp
#include "FileHandler.h"  // Include the header file
#include <stdexcept>      // For runtime_error exceptions
#include <sstream>        // For string streams

// Create a file at the specified path
void FileHandler::createFile(const std::string& filePath) {
    std::ofstream file(filePath);  // Open file stream for writing
    if (!file) {  // Check if the file was created successfully
        throw std::runtime_error("Unable to create file: " + filePath);  // Throw error if not
    }
    file.close();  // Close the file
}

// Write content to a file at the specified path
void FileHandler::writeFile(const std::string& filePath, const std::string& content) {
    std::ofstream file(filePath);  // Open file stream for writing
    if (!file) {  // Check if the file was opened successfully
        throw std::runtime_error("Unable to open file: " + filePath);  // Throw error if not
    }
    file << content;  // Write content to the file
    file.close();  // Close the file
}

// Read content from a file at the specified path
std::string FileHandler::readFile(const std::string& filePath) {
    std::ifstream file(filePath);  // Open file stream for reading
    if (!file) {  // Check if the file was opened successfully
        throw std::runtime_error("Unable to open file: " + filePath);  // Throw error if not
    }
    std::ostringstream ss;
    ss << file.rdbuf();  // Read the file content into the string stream
    return ss.str();  // Return the file content as a string
}
